### PR TITLE
EIP-6110: Supply validator deposits on chain

### PR DESCRIPTION
**Pull request by gumb0**
https://eips.ethereum.org/EIPS/eip-6110

### CODE REVIEW

File: test/state/requests.hpp [hunk 1]
```diff
@@ -37,9 +38,15 @@ struct Requests
     Type type = Type::deposit;
     /// Requests data - an opaque byte array, contains zero or more encoded request objects.
     evmc::bytes data;
+
+    explicit Requests(Type _type, evmc::bytes _data = {}) noexcept
```
Comment (gumb0) [inline test/state/requests.hpp:42]: Needed to make `vector::emplace_back()` work for this struct on some compilers.

Comment (chfast) [reply test/state/requests.hpp:42]: I use other workaround: `emplace_back(Requests{...})`. I wonder which pattern is easier to find later...

File: test/state/requests.cpp [hunk 2]
```diff
@@ -30,4 +36,56 @@ hash256 calculate_requests_hash(std::span<const Requests> block_requests_list)
...
+                //     bytes pubkey,
+                //     bytes withdrawal_credentials,
+                //     bytes amount,
+                //     bytes signature,
+                //     bytes index
+                // );
+                //
+                // In ABI every bytes array data is prepended by a word with its size.
+                // Skip over first 5 words (offsets of the values) and over pubkey size.
+                constexpr auto pubkey_offset = 32 * 5 + 32;
```
Comment (chfast) [inline test/state/requests.cpp:61]:
```suggestion
                static constexpr auto PUBKEY_OFFSET = 32 * 5 + 32;
```

File: test/state/requests.hpp (hunk 1 above)
Comment (gumb0) [reply test/state/requests.hpp:42]: Usually me too, but this time I think I got error from clang-tidy "unnecessary temporary object created"

File: test/state/requests.cpp (hunk 2 above)
Comment (chfast) [reply test/state/requests.cpp]: upper case?

**Review by chfast — APPROVED** [review]

**Review by pdobacz — APPROVED** [review]

**Pull request merged**